plugins/inputs/http/http.go

```diff
@@ -0,0 +1,165 @@
+package http
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+)
+
+type HTTP struct {
+	URLs []string `toml:"urls"`
+
+	// HTTP Basic Auth Credentials
+	Username string
+	Password string
+
+	// Path to CA file
+	SSLCA string `toml:"ssl_ca"`
+	// Path to host cert file
+	SSLCert string `toml:"ssl_cert"`
+	// Path to cert key file
+	SSLKey string `toml:"ssl_key"`
+	// Use SSL but skip chain & host verification
+	InsecureSkipVerify bool
+
+	Timeout internal.Duration
+
+	client *http.Client
+
+	// The parser will automatically be set by Telegraf core code because
+	// this plugin implements the ParserInput interface (i.e. the SetParser method)
+	parser parsers.Parser
+}
+
+var sampleConfig = `
+  ## One or more URLs from which to read formatted metrics
+  urls = [
+    "http://localhost:2015/simple.json"
+  ]
+
+  ## Optional HTTP Basic Auth Credentials
+  # username = "username"
+  # password = "pa$$word"
+
+  ## Optional SSL Config
+  # ssl_ca = "/etc/telegraf/ca.pem"
+  # ssl_cert = "/etc/telegraf/cert.pem"
+  # ssl_key = "/etc/telegraf/key.pem"
+  ## Use SSL but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## http request & header timeout
+  ## defaults to 5s if not set
+  timeout = "10s"
+
+  ## Mandatory data_format
+  ## See available options at https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
+  data_format = "json"
+`
+
+// SampleConfig returns the default configuration of the Input
+func (*HTTP) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns a one-sentence description on the Input
+func (*HTTP) Description() string {
+	return "Read formatted metrics from one or more HTTP endpoints"
+}
+
+// Gather takes in an accumulator and adds the metrics that the Input
+// gathers. This is called every "interval"
+func (h *HTTP) Gather(acc telegraf.Accumulator) error {
+	if h.client == nil {
+		tlsCfg, err := internal.GetTLSConfig(
+			h.SSLCert, h.SSLKey, h.SSLCA, h.InsecureSkipVerify)
+		if err != nil {
+			return err
+		}
+		h.client = &http.Client{
+			Transport: &http.Transport{
+				ResponseHeaderTimeout: h.Timeout.Duration,
```

Comment:
Don't set a response header timeout since we have the client timeout.  I need to exercise these from the code so they stop getting copied around.